server/src/main/java/com/github/sgov/server/service/repository/WorkspaceRepositoryService.java

```diff
@@ -134,4 +162,19 @@ public void loadContext(final VocabularyContext vocabularyContext) {
             e.printStackTrace();
         }
     }
+
+    /**
+     * Loads all workspaces including all its inferred properties.
+     *
+     * @return List of all workspaces
+     */
+    public List<Workspace> findAllInferred() {
+        List<Workspace> workspaces = findAll();
+        List<VocabularyContext> vocabularyContexts = workspaces.stream()
+            .map(Workspace::getVocabularyContexts)
+            .flatMap(Set::stream)
+            .collect(Collectors.toList());
+        workspaceDao.setVocabularyLabels(vocabularyContexts, "cs");
```

Comment:
Takže labely budeme setovat při čtení touto metodou? Pokud musí být v datech, neměli bychom to dělat při ukládání?

Reply:
1)  "Takže labely budeme setovat při čtení touto metodou?"
Ano, setVocabularyLabels, ale nic neperzistuje. Jestli je problem v neituitivnosti metody setVocabularyLabels, muzu rozdelit metodu na getVocabularyLabels a setovani provadet po jeji zavolani.

2) reseni se ani mne moc nelibi ... inference v repozitari mi pride lepsi, chapu vsak, ze nechces zakladat pravidla, kdyz je zatim nepotrebujeme.

3) "Pokud musí být v datech ...?" 
Nepotrebujeme to mit v datach, potrebujeme to jen v REST API. @Inferred jsem pouzil na JOPA objektu jenom proto, ze se mi nechtelo psat nove DTO a chtel jsem to mit read-only. 

4) "... neměli bychom to dělat při ukládání?" 
No mne by nejvic vadil tento scenar: TermIt si otevre slovnik a modifikuje nektere metadata slovniku vcetne label-u. Pak slovnik ulozi a take by mel povinnost ulozit tento label do metadatoveho kontextu. To jiste plati o OntoGrapher-u.